spectral_cube/spectral_cube.py

```diff
@@ -1132,7 +1132,7 @@ def percentile(self, q, axis=None, iterate_rays=False, **kwargs):
 
         return result
 
-    def with_mask(self, mask, inherit_mask=True, wcs_tolerance=None):
+    def with_mask(self, mask, inherit_mask=True, wcs_tolerance=None, **kwargs):
```

Comment:
I'm not sure if this is the best approach, since it adds **kwargs to the public API, but otherwise we'd need to repeat some code.

Reply:
I think that's fine, though it might result in docs headaches later...

Reply:
Removed - I've tried a different way now